lib/chai/core/assertions.js

```diff
@@ -235,55 +346,144 @@ module.exports = function (chai, _) {
-   * the inclusion of a value in an array, a substring in a string, or a subset
-   * of properties in an object.
+   * The `deep` and `own` flags can be combined when the target is an object.
    *
-   *     expect([1,2,3]).to.include(2);
-   *     expect('foobar').to.contain('foo');
-   *     expect({ foo: 'bar', hello: 'universe' }).to.include({ foo: 'bar' });
+   *     expect({a: {b: 2}}).to.deep.own.include({a: {b: 2}});
    *
-   * By default, strict equality (===) is used. When asserting the inclusion of
-   * a value in an array, the array is searched for an element that's strictly
-   * equal to the given value. When asserting a subset of properties in an
-   * object, the object is searched for the given property keys, checking that
-   * each one is present and stricty equal to the given property value. For
-   * instance:
+   * If the `nested` flag is set when the target is an object, dot- and
+   * bracket-notation can be used to reference nested properties.
    *
-   *     var obj1 = {a: 1}
-   *       , obj2 = {b: 2};
-   *     expect([obj1, obj2]).to.include(obj1);
-   *     expect([obj1, obj2]).to.not.include({a: 1});
-   *     expect({foo: obj1, bar: obj2}).to.include({foo: obj1});
-   *     expect({foo: obj1, bar: obj2}).to.include({foo: obj1, bar: obj2});
-   *     expect({foo: obj1, bar: obj2}).to.not.include({foo: {a: 1}});
-   *     expect({foo: obj1, bar: obj2}).to.not.include({foo: obj1, bar: {b: 2}});
+   *     expect({a: {b: ['x', 'y']}}).to.nested.include({'a.b[1]': 'y'});
    *
-   * If the `deep` flag is set, deep equality is used instead. For instance:
+   * If `.` or `[]` are part of an actual property name, they can be escaped by
+   * adding two backslashes before them.
    *
-   *     var obj1 = {a: 1}
-   *       , obj2 = {b: 2};
-   *     expect([obj1, obj2]).to.deep.include({a: 1});
-   *     expect({foo: obj1, bar: obj2}).to.deep.include({foo: {a: 1}});
-   *     expect({foo: obj1, bar: obj2}).to.deep.include({foo: {a: 1}, bar: {b: 2}});
+   *     expect({'.a': {'[b]': 2}}).to.nested.include({'\\.a.\\[b\\]': 2});
    *
-   * These assertions can also be used as property based language chains,
-   * enabling the `contains` flag for the `keys` assertion. For instance:
+   * The `deep` and `nested` flags can be combined when the target is an object.
    *
-   *     expect({ foo: 'bar', hello: 'universe' }).to.include.keys('foo');
+   *     expect({a: {b: [{c: 3}]}}).to.deep.nested.include({'a.b[0]': {c: 3}});
+   *
+   * The `own` and `nested` flags cannot be combined.
+   *
+   * `.include` can be negated by adding `.not` earlier in the assertion chain.
+   *
+   *     expect('foobar').to.not.include('taco');
+   *     expect([1, 2, 3]).to.not.include(4);
+   * 
+   * However, `.include` should rarely be negated when the target is an object.
+   * The problem is that it creates uncertain expectations by asserting that the
+   * target object doesn't have all of `val`'s properties but may or may not
```

Comment:
💭 It's actually worse than that, it will assert that the target doesn't have those properties _with those values_. 

```js
expect({ c: 3 }).to.not.include({ c: 4 }) // passes
```